readers/api/endpoint.go

```diff
@@ -27,10 +27,40 @@ func ListChannelMessagesEndpoint(svc readers.MessageRepository) endpoint.Endpoin
 			return nil, err
 		}
 
-		return listChannelMessagesPageRes{
+		return listMessagesPageRes{
 			PageMetadata: page.PageMetadata,
 			Total:        page.Total,
 			Messages:     page.Messages,
 		}, nil
 	}
 }
+
+func listAllMessagesEndpoint(svc readers.MessageRepository) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listMessagesReq)
+		if err := req.validateWithNoChanel(); err != nil {
+			return nil, err
+		}
+
+		// Check if user is authorized to read all messages
+		if err := authorizeAdmin(ctx, "authorities", "member", req.token); err != nil {
+			return nil, errors.Wrap(errors.ErrAuthentication, err)
```

Comment:
This should be `errors.Wrap(errors.ErrAuthorization, err)` but check what the method is returning, maybe there is no need to Wrap it